src/Microsoft.AspNetCore.Mvc.Analyzers/CodeAnalysisExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Mvc.Analyzers
+{
+    internal static class CodeAnalysisExtensions
+    {
+        public static bool HasAttribute(this ITypeSymbol typeSymbol, ITypeSymbol attribute, bool inherit)
```

Comment:
I'm not a fan of extensions in general but these seem like real gaps in Roslyn's api - unless this code is wrong